test/e2e/framework/pod/wait.go

```diff
@@ -211,13 +218,83 @@ func WaitForPodsRunningReady(ctx context.Context, c clientset.Interface, ns stri
 	}))
 
 	// An error might not be fatal.
-	if err != nil && notReady >= 0 && notReady <= allowedNotReadyPods {
-		framework.Logf("Number of not-ready pods (%d) is below the allowed threshold (%d).", notReady, allowedNotReadyPods)
+	if nOk+len(otherPods) >= minPods && len(otherPods) <= allowedNotReadyPods {
 		return nil
 	}
 	return err
 }
 
+// WaitForPodsRunningReady waits up to timeout for the following conditions:
+//  1. At least minPods Pods in Namespace ns are Running and Ready
+//  2. No Pods in Namespace ns are not either Ready, Succeeded, or Failed with a Controller
```

Comment:
This second criteria is hard to understand (double negation, what is "Failed with a Controller"). How about this?

```suggestion
//  2. Only Pods owned by a controller are allowed to be in the Failed phase.
// 3. Pods in the Succeeded phase are ignored.
// 4. No Pod is in some other phase.
```

Reply:
I definitely agree it's hard to understand. I was originally worried about another Pod Phase being added and making this comment absolete, but that's far too unlikely to be worth the convoluted English.

I changed it to `No Pods in Namespace ns are Failed and not owned by a controller or Pending`, since Pending's also an important case. Let me know if you'd like me to update it again.